tests/garage/tf/models/test_new_model_pickle.py

```diff
@@ -0,0 +1,63 @@
+import pickle
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+
+
+class TestNewModelPickling(unittest.TestCase):
```

Comment:
This can be a different test case in `test_model.py`

Reply:
We usually try to keep a 1:1 mapping between source units and test files.

Reply:
I am having issue when I called `reset_default_graph()`. I guess the unit tests are run in parallel and somehow when `reset_default_graph()` is called, it is in another `tf.Session` from another unit test.

Reply:
Em……I don't think it is true that when you call `reset_default_graph` it is called in another session. This doesn't make sense. Why don't use TfGraphTest?

Reply:
When I run the above tests standalone, it is fine. When I put it under `test_new_model.py` and run, it said I am calling `reset_default_graph()` within a session, which I didn't.

Reply:
Is test_new_model running under the TfGraphTest? Then there is a session created in the setUp.

Reply:
but in principle, each setUp() is called before each test? If the tests are done one by one, it should be fine.

Reply:
I don't think it would be fine. `setUp` is called before the test is called. So there is a session created.

Reply:
What do you mean by `put it under test_new_model.py`?

Reply:
I first tried to put the pickle test under `TfGraphTest` and I did not create new session. I just used the one created in setUp().

Reply:
I think by `put it under test_new_model.py`, the class should become a test case under the TestModel class. In this case, if `tf.reset_default_graph()` is called then there will be an error because of attempting to reset graph when a session is active. 

I suppose you need to call `reset_default_graph()` because unpickle uses the same variable scope. Usually calling the function in unittest is unpredictable because multiple graphs may be created and it is hard to know which one is default.

There are multiple ways to do this without calling it. 
`with tf.Graph().as_default()` or `sess = tf.Session(tf.Graph())`

Reply:
Yes, you are right. `with tf.Session(graph=tf.Graph()) as sess:` does the job.